ee/management/commands/backfill_persons_and_groups_on_events.py

```diff
@@ -0,0 +1,148 @@
+Also note that minor inconsitencies can occur from distinct IDs and properties changing during the backfill.
+
+3. Drop the dictionaries
+"""
+
+logger = structlog.get_logger(__name__)
+
+GROUPS_DICT_TABLE_NAME = f"{GROUPS_TABLE}_dict"
+PERSONS_DICT_TABLE_NAME = f"{PERSONS_TABLE}_dict"
+PERSON_DISTINCT_IDS_DICT_TABLE_NAME = f"{PERSON_DISTINCT_ID2_TABLE}_dict"
+
+GROUPS_DICTIONARY_SQL = f"""
+CREATE DICTIONARY IF NOT EXISTS {GROUPS_DICT_TABLE_NAME}
+ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+(
+    group_key String,
+    group_properties String
+)
+PRIMARY KEY group_key
+SOURCE(CLICKHOUSE(TABLE {GROUPS_TABLE} DB '{CLICKHOUSE_DATABASE}' USER 'default'))
+LAYOUT(cache(size_in_cells 1000))
+Lifetime(60000)
+"""
+
+
+PERSON_DISTINCT_IDS_DICTIONARY_SQL = f"""
+CREATE DICTIONARY IF NOT EXISTS {PERSON_DISTINCT_IDS_DICT_TABLE_NAME}
+ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+(
+    distinct_id String,
+    person_id UUID
+)
+PRIMARY KEY distinct_id
+SOURCE(CLICKHOUSE(TABLE person_distinct_id DB '{CLICKHOUSE_DATABASE}' USER 'default'))
+LAYOUT(cache(size_in_cells 1000))
+Lifetime(60000)
+"""
+
+PERSONS_DICTIONARY_SQL = f"""
+CREATE DICTIONARY IF NOT EXISTS {PERSONS_DICT_TABLE_NAME}
+ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+(
+    id UUID,
+    properties String
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE person DB '{CLICKHOUSE_DATABASE}' USER 'default'))
+LAYOUT(cache(size_in_cells 1000))
+Lifetime(60000)
+"""
+
+BACKFILL_BASE_SQL = f"""
+ALTER TABLE {EVENTS_DATA_TABLE()}
+ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+UPDATE
+    person_id=toUUID(dictGet('{PERSON_DISTINCT_IDS_DICT_TABLE_NAME}', 'person_id', distinct_id)),
+    person_properties=dictGetString('{PERSONS_DICT_TABLE_NAME}', 'properties', toUUID(dictGet('{PERSON_DISTINCT_IDS_DICT_TABLE_NAME}', 'person_id', distinct_id))),
+    group0_properties=dictGetString('{GROUPS_DICT_TABLE_NAME}', 'group_properties', $group_0),
+    group1_properties=dictGetString('{GROUPS_DICT_TABLE_NAME}', 'group_properties', $group_1),
+    group2_properties=dictGetString('{GROUPS_DICT_TABLE_NAME}', 'group_properties', $group_2),
+    group3_properties=dictGetString('{GROUPS_DICT_TABLE_NAME}', 'group_properties', $group_3),
+    group4_properties=dictGetString('{GROUPS_DICT_TABLE_NAME}', 'group_properties', $group_4)
+"""
+
+DROP_GROUPS_DICTIONARY_SQL = f"DROP DICTIONARY IF EXISTS {GROUPS_DICT_TABLE_NAME}"
+DROP_PERSON_DISTINCT_IDS_DICTIONARY_SQL = f"DROP DICTIONARY IF EXISTS {PERSON_DISTINCT_IDS_DICT_TABLE_NAME}"
+DROP_PERSONS_DICTIONARY_SQL = f"DROP DICTIONARY IF EXISTS {PERSONS_DICT_TABLE_NAME}"
+
+query_number = 0
+
+
+def print_and_execute_query(sql, name, dry_run, timeout=180):
+    global query_number
+
+    print(f"> {query_number}. {name}", end="\n\n")
+    print(sql, end="\n")
+    print("---------------------------------", end="\n\n")
+
+    query_number = query_number + 1
+
+    if not dry_run:
+        sync_execute(sql, {"max_execution_time": timeout, "max_memory_usage": 1000000000000})
```

Comment:
We should set max_execution_time to 0 (unlimited). Timing these out will do no good.